subprojects/core/src/main/java/org/gradle/internal/fingerprint/AbsolutePathInputNormalizer.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.fingerprint;
+
+import org.gradle.api.internal.changedetection.state.FileCollectionSnapshotter;
+import org.gradle.api.tasks.FileNormalizer;
+
+/**
+ * {@link FileNormalizer} that uses absolute paths for input files. The default.
+ */
+public interface AbsolutePathInputNormalizer extends FileCollectionSnapshotter, FileNormalizer {
+}
```

Comment:
I don't think this makes sense to implement `FileCollectionSnapshotter`: we don't ever use it as such. Instead, we should make this public API as a `FileNormalizer`. Same for all the other new `FileNormalizer`s.

Reply:
I separated the fingerprinter interfaces from the normalizer interfaces.

I wouldn't make this public API right now, I don't see a reason why someone would want to use those `Normalizer`s. WDYT?

Reply:
I think it's fine for now.